exes/mqtt-bridge/inc/tfc_to_external.hpp

```diff
@@ -75,6 +77,17 @@ class tfc_to_external {
         [this](std::vector<ipc_ruler::signal> const& signals) { handle_incoming_signals_from_ipc_client(signals); });
```

Comment:
line 76 here is doing the same work as the static call in schema, you can expose the output of this call to thread_local vector, but remember to put it into source file similar to this https://github.com/Skaginn3x/framework/blob/main/libs/testing/src/asio_clock.cpp#L6 and expose it via setter and getter

Then you won't need to create the context and block, just use the getter function

Reply:
I don't understand, the struct still needs to be initialized.

In the clock example it is initialized as such:

```cpp
asio::basic_waitable_timer<tfc::testing::clock> timer{ ctx };
```

making each instance a new instance, and therefore has the same effect as just calling the signals callback on the ipc client.

Reply:
I suggest new header but you can use this one,

in header make free functions
```cpp
namespace global { // please name better
void set_signal_names(std::vector<std::string> const&);
std::vector<std::string> const& get_signal_names();
}
```

source file
```cpp
namespace {
thread_local std::vector<std::string> signal_names{};
}
namespace global { // please name better
void set_signal_names(std::vector<std::string> const& names) {
  signal_names = names;
}
std::vector<std::string> const& get_signal_names() {
  return signal_names;
}
}
```

working example https://godbolt.org/z/73drvvPan

Reply:
Then the following code can be simplified: 
```cpp
template <>
struct tfc::json::detail::to_json_schema<signal_name> {
  template <auto Opts>
  static void op(auto& s, auto& defs) noexcept {
    for (auto const& name : global::get_signal_names()) {
      s.oneOf.push_back(tfc::json::detail::schematic{ .attributes{ tfc::json::schema{ .title = name } } });
    };
  }
};
```

Reply:
and `handle_incoming...` could call `global::set_signal_names(...)`